monkey/monkey_island/cc/services/infection_lifecycle.py

```diff
@@ -10,9 +10,8 @@
 logger = logging.getLogger(__name__)
 
 
-def get_completed_steps():
-    is_any_exists = NodeService.is_any_monkey_exists()
-    infection_done = NodeService.is_monkey_finished_running()
+def get_completed_steps(agent_repository: IAgentRepository):
+    infection_done = _is_infection_done(agent_repository)
```

Comment:
If we do this then what's the point of changes in `Main.tsx`?

Reply:
I think `main.tsx` overrides `infection_done` step, I don't think we need to correct it in the back end

Reply:
The infection lifecycle still uses it for report generation.